iot-hub/Samples/device/MessageReceiveSample/MessageReceiveSample.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.Client.Samples
+{
+    public class MessageReceiveSample
```

Comment:
meaningful doc comments please

Reply:
For me it would be about illustrating the 2 different ways of receiving messages, through polling or a callback, how to receive messages both ways, and how to change between them. We should also talk about why one may want one approach over the other, and how generally we wouldn't expect a device app to actively switch.

Reply:
updated